jupyterhub/schema.yaml

```diff
@@ -852,6 +852,12 @@ properties:
                 type: integer
                 description: |
                   The HTTPS port the proxy-public service should be exposed on.
+          disableHttpPort:
+            type: boolean
+            description: |
+              Default `false`.
+
+              If `true` disables service HTTP traffic on port 80, and therefore forcing use of HTTPS.
```

Comment:
```suggestion
              If `true`, port 80 for incoming HTTP traffic will no longer be exposed. This should not be used with `proxy.https.type=letsencrypt` or `proxy.https.enabled=false` as it would remove the only exposed port.
```

Reply:
> This should not be used with `proxy.https.type=letsencrypt`
I don't think this is true
Even though you would use letsencrypt and proxy to handle TLS termination, you still want public Load balancer to listen on port 443 (not 80) .. because that's the port the browser hits when you prefix it with `https://`
If you use letsencrypt, you just don't add SSL annotations on the Service, so that it just forwards the traffic, without decrypting it.

The only case when you don't want to disable port 80 is, when either you don't want to use HTTPS at all, or when the upstream proxy is correctly handling redirection from http to https (but disabling port on the load balancer, not the proxy itself .. proxy always listens on only one port, if it doesn't handle TLS termination)
https://aws.amazon.com/premiumsupport/knowledge-center/redirect-http-https-elb/

Reply:
If you configure `proxy.https.type=letsencrypt` (assuming also `proxy.https.enabled`), then I'm very confident you are required to have port 80 exposed. If you don't, then the ACME procedure using a HTTP01 challenge to acquire a new certificate that we have setup as part of this Helm chart would fail.

Reply:
Ah you are right. Forgot about the http challenge
But if you leave the 80 port open .. does it also forward traffic to the jupyterhub application? because if so, in this case it should handling redirect to https

Reply:
There will be a redirect from the `autohttps` pod unless the HTTP traffic was associated with an ACME challenge.

This setup is very complicated in general, and makes things hard to understand. The proxy-public k8s Service will redirect to the autohttps pod which will redirect to a k8s Service named `proxy-http` after TLS termination. So the same k8s Service can go to different pods depending on the configuration of the Helm chart. Either it goes directly to the `proxy` pod, or it goes through the `autohttps` pod first which in turn will redirect non ACME http requests to HTTPS.